// @aws-sdk/client-dynamodb/dist-es/commands/ExecuteStatementCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ExecuteStatementInputFilterSensitiveLog, ExecuteStatementOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_0ExecuteStatementCommand, serializeAws_json1_0ExecuteStatementCommand, } from "../protocols/Aws_json1_0";
export class ExecuteStatementCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ExecuteStatementCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "DynamoDBClient";
        const commandName = "ExecuteStatementCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ExecuteStatementInputFilterSensitiveLog,
            outputFilterSensitiveLog: ExecuteStatementOutputFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_0ExecuteStatementCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_0ExecuteStatementCommand(output, context);
    }
}
